contracts/StorageMarket.sol

```diff
@@ -94,64 +112,84 @@ contract StorageMarketPlace is Sablier {
+            "amount value cannot be less than value of 1 day"
+        );
+        require(_minDurationInDays >= 1, "minDuration should be atleast 1 day");
+        SubscriptionInfo storage subscription = subscriptions[msg.sender];
+        subscription.isEnabled = true;
+        subscription.minDurationInDays = _minDurationInDays;
+        subscription.amountPerDay = _amountPerDay;
+        subscription.tokenAddress = _tokenAddress;
+        emit SubscriptionInfoUpdated(msg.sender, true);
+    }
+
+    function disableSubscriptionInfo() external {
+        SubscriptionInfo storage subscription = subscriptions[msg.sender];
+        subscription.isEnabled = false;
+        emit SubscriptionInfoUpdated(msg.sender, false);
+    }
+
+    /**
+     * @dev Creates a Subscription for future content for that particular seller during a specific duration
+     * @param _deposit - Subscription amount
+     * @param _token - payment asset address
+     * @param _numDays - number of days of subscription
+     * @param _seller - seller address whose subscription is being brought
+     */
     function createSubscription(
         uint256 _deposit,
         address _token,
-        uint256 _startTime,
-        uint256 _stopTime,
-        address reciever
+        uint256 _numDays,
+        address _seller
     ) external {
-        require(reciever != address(0x00), "cannot stream to the zero address");
-        require(
-            reciever != address(this),
-            "cannot stream to the contract itself"
-        );
-        require(reciever != msg.sender, "cannot stream to yourself");
-        require(_deposit > 0, "deposit is zero");
-        require(
-            _startTime > block.timestamp,
-            "start time should be after block.timestamp"
-        );
-        require(_stopTime > _startTime, "start time has to be before stoptime");
-        uint256 delta = _stopTime.sub(_startTime);
+        SubscriptionInfo storage subscription = subscriptions[_seller];
+        uint256 _oldStreamId = subscription.streams[msg.sender];
+        require(subscription.isEnabled, "Seller has not enabled subscriptions");
+        require(!isValid(_oldStreamId), "Buyer already holders a subscription");
         require(
-            _deposit > delta,
-            "Deposit should be more than delta as specified by EIP-1620"
+            _deposit == _numDays.mul(subscription.amountPerDay),
+            "Deposit amount incorrect"
         );
-        // THE Official implementation of the eip-1620 specifies that the deposit shopuld be a multiple of the delta just for the payament rate, the difference is negligible to be calculated efficiently specified in eip-1620 docs
-        // Check EIP-1620 for more details
-        uint256 remainder = _deposit.mod(delta);
-        uint256 actualDepositAmount = _deposit.sub(remainder);
-        uint256 streamId = createStream(
-            reciever,
-            actualDepositAmount,
+        uint256 _startTime = block.timestamp;
```

Comment:
without delta logic, the stream creation has chance to fail this is needed since it is checked in their logic